Source/ASDisplayNode.mm

```diff
@@ -1546,78 +1547,87 @@ - (void)_updateClipCornerLayerContentsWithRadius:(CGFloat)radius backgroundColor
 {
   ASPerformBlockOnMainThread(^{
     for (int idx = 0; idx < NUM_CLIP_CORNER_LAYERS; idx++) {
-      // Layers are, in order: Top Left, Top Right, Bottom Right, Bottom Left.
+      // Skip corners that aren't clipped (we have already set up & torn down layers based on maskedCorners.)
+      if (_clipCornerLayers[idx] == nil) {
+        continue;
+      }
+
+      // Layers are, in order: Top Left, Top Right, Bottom Left, Bottom Right, which mirrors CACornerMask.
       // anchorPoint is Bottom Left at 0,0 and Top Right at 1,1.
       BOOL isTop   = (idx == 0 || idx == 1);
-      BOOL isRight = (idx == 1 || idx == 2);
-      
+      BOOL isRight = (idx == 1 || idx == 3);
```

Comment:
The reason some of these bits flipped is (1) I changed our NUM_CLIP_CORNERS array order to make the CACornerMask order and (2) the original comments about CA being inverted in this context are incorrect.